blog/2015/04/quality-is-free-getting-there-isnt.html

```diff
@@ -17,13 +17,52 @@
   The journal is also exploring whether peer reviewers can test complex code..."
   That's a welcome step in theory,
   but I worry about how it will play out in practice.
-  Scientists already complain about how much time they spend reviewing papers.
-  If they have to review code as well
+  Scientists already complain about how much time they spend reviewing papers:
+  reviewing code as well will take even more time,
+  particularly if:
 </p>
 <!-- end excerpt -->
+<ul>
+  <li>
+    they haven't been taught how to do reviews, and
+  </li>
+  <li>
+    the code hasn't been written with review in mind.
+  </li>
+</ul>
 <p>
-  Chris Mentzel
-  (a program officer with the Moore Foundation)
-  <a href="https://twitter.com/moorescientific/status/566645532018229249">tweeted</a> back in February that,
-  "Success in data science will require championing education and training at all levels, with a focus on people."
+  As we found in <a href="http://arxiv.org/abs/1407.5648">our two-part study</a>
+  of code review by and for scientists,
+  most groups and software currently fail both of these tests:
+  they don't know what to look for in code reviews because they've never taken part in one,
+  and because they don't know what <em>other</em> people will be looking for,
+  they don't make things easy to find and understand in their own work.
+  I therefore believe that asking the average scientist to review code will take more time than it delivers value.
+</p>
+<p>
+  But this can be fixed.
+  Teaching people how to write readable code is a first step,
+  and that's always been part of Software Carpentry's mission.
+  Showing them what code reviews look like&mdash;e.g., doing one live in front of the class&mdash;hasn't been part of our workshops,
```

Comment:
@abostroem Did we ever write anything about doing this at AAS or the other workshop where someone did code review? Did we ever publish this post? https://github.com/abostroem/2015-01-03-aas/blob/gh-pages/blog_post/swc_aas_notes.md I don't think we did.

Link to the code review sample is at: https://github.com/abostroem/2015-01-03-aas/tree/gh-pages/novice/code-review